app/models/service/dialog_properties/retirement.rb

```diff
@@ -51,7 +48,10 @@ def retire_on_date=(value)
 
       def retirement_warning
         warn_value = parse_retirement_warn
-        @attributes[:retirement_warn] = warn_value if warn_value
+        if warn_value
+          days_between_warn_and_retirement = (@attributes[:retires_on] - warn_value) / 1.day.seconds.to_i
```

Comment:
You get the same results with just `1.day` without the `.seconds.to_i`.

Doing the subtraction on the Time classes results in a float and then the whole operation returns a float for `retirement_warn`.  Since we are expecting an integer value it would be better to adjust that as well.

`(@attributes[:retires_on] - warn_value).to_i / 1.day`

Otherwise this looks good.